test/files/neg/t10507.check

```diff
@@ -0,0 +1,4 @@
+Test_2.scala:2: error: called constructor's definition must precede calling constructor's definition
```

Comment:
Rare case where spec wording is clearer: "To prevent infinite cycles of constructor invocations, there is the restriction that every self constructor invocation must refer to a constructor definition which precedes it."

I might word it: "As a restriction to prevent  infinite cycles of constructor invocations, every self constructor invocation must refer to a constructor definition which precedes it."

Reply:
Not sure, are you asking to change the wording in this PR or not? 🤔